projects/js-packages/tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
+	"extends": "../../tools/js-tools/tsconfig.base.json",
+	"compilerOptions": {
+		"baseUrl": ".",
+		"paths": {
+			"*": [
+				"*",
+				// fallback to resolving imports relative to /tools/*
+				"../../tools/js-tools/node_modules/*",
+				"../../tools/js-test-runner/node_modules/*"
```

Comment:
> Why is any typescript code even trying to load anything from either of those locations?

No TypesScript code is trying to load anything from those two locations. It's just that TS can't find `@testing-library/react` installed in the package and by using these paths in tsconfig, TS will not look for these in the package itself, rather it look inside `/tools/*`.

> I see another comment mentioned something not finding `@testing-library/react`. If that's because something is using `js-test-runner`, it would probably be better to migrate away from that and use jest directly.

I don't think that will make any difference. It's just that when it comes to TS, it's strict about finding dependencies. Though we can manipulate the resolution as done here in compiler paths.

These paths here are nothing but a way to tell TS where to find the modules if it can't find them installed in the package.

Reply:
> No TypesScript code is trying to load anything from those two locations.

Apparently something is, or why are you messing with this in the first place?

> I don't think that will make any difference.

It should. If typescript can't find _directly installed_ dependencies then it's fundamentally broken. But I doubt that's the case.

Reply:
>>>> I see another comment mentioned something not finding `@testing-library/react`. If that's because something is using `js-test-runner`, it would probably be better to migrate away from that and use jest directly (and then also dev-depend on `@testing-library/react` directly).
>>>
>>> I don't think that will make any difference.
>>
>> It should. If typescript can't find directly installed dependencies then it's fundamentally broken. But I doubt that's the case.
>
> That's the problem, the dependency is NOT directly installed, 

Directly installing the dependency is exactly what we've been talking about here.

Reply:
> Directly installing the dependency is exactly what we've been talking about here.

Alright, is installing the dependency (`@testing-library/react`) directly in `js-packages/components` fine?

If that's the case then there is no problem at all. TypesScript will be happy and we won't need this tsconfig here.

Reply:
It's fine with me. :+1:

Reply:
Thank you. Actually it was after @renatoagds suggested here that I had to make these changes in order to apply that suggetion.
https://github.com/Automattic/jetpack/pull/23889#discussion_r852495895

Alright, I'll revert the lastest changes and install the dependency directly into js-packages/components.

Reply:
Thanks for pointing out all this @anomiex ... Let's move this way @manzoorwanijk 🚀

Reply:
Alright, I have reverted the changes. Thank you for your feedback.